src/sp/it/pl/util/system/Environment.kt

```diff
@@ -260,7 +260,7 @@ fun openIn(files: List<File>) {
     }
 }
 
-fun chooseFile(title: String, type: FileType, initial: File?, w: Window?, vararg extensions: FileChooser.ExtensionFilter): Try<File, Void> {
+fun chooseFile(title: String, type: FileType, initial: File? = null, w: Window? = null, vararg extensions: FileChooser.ExtensionFilter): Try<File, Void> {
```

Comment:
I have checked all usages of these methods, and found that (as exepcted), this change is meaningless.

Reply:
But is there any downside to it? it can ease further usages.

Reply:
Yes
- APi gets more complex to understand.
- It may lead to future bugs, because callees effectively depend on the default argument values, which however can be changed by an API maintainer - this may have great consequences.
- Default arguments are basically the equivalent to method overloading, which is good to avoid if possible. Using multiple default arguments forces the callee to use named argument syntax or lose any gain.

So use default args only when the gain is justified. This is NOT the case. In fact no single call within the application even needs it! I'm leaving it as is, as a little experiment.